subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/CurrentJvmToolchainSpec.java

```diff
@@ -25,6 +25,11 @@ public class CurrentJvmToolchainSpec extends DefaultToolchainSpec {
     public CurrentJvmToolchainSpec(ObjectFactory factory) {
         super(factory);
         getLanguageVersion().set(JavaLanguageVersion.of(Jvm.current().getJavaVersion().getMajorVersion()));
+
+        // disallow changing property values
+        getLanguageVersion().finalizeValue();
+        getVendor().finalizeValue();
+        getImplementation().finalizeValue();
```

Comment:
I am not sure why this was allowed previously, because changing the properties on the `CurrentJvmToolchain` can completely mess up caching.